connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -348,7 +347,7 @@ private void commitOffsets(long now, boolean closing) {
 
         if (commitableOffsets.equals(lastCommittedOffsets)) {
             log.debug("{} Skipping offset commit, no change since last commit", this);
-            onCommitCompleted(null, commitSeqno);
+            onCommitCompleted(null, commitSeqno, null);
```

Comment:
No need to fix this here since it would be risky to change, but don't you think it's a little odd that we increment the sequence number even if nothing has changed? I wonder if it would be more intuitive to bump the sequence only prior to the call to `doCommit` below.